fix(tokens): DLT-2058 fix uncalced focus ring token

braddialpad opened the pull request:
# fix(tokens): DLT-2058 fix uncalced focus ring token

<!--- Feel free to remove any unused sections -->

## Obligatory GIF (super important!)

<!--- go to giphy.com, pick a gif, click share, copy gif link, paste within the () brackets below. -->

![Obligatory GIF](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExY2I0NWJsaGRlengzenl6NndtcDJpanlsdGh3aGxsZW01OTYwdmI5diZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/CLQNFY6YIHanjWSEhW/giphy.gif)

## :hammer_and_wrench: Type Of Change

<!--- Tick or place an `x` for the type of change. Should match the type in the commit message / PR title
in https://github.com/dialpad/dialtone/blob/staging/.github/COMMIT_CONVENTION.md -->

These types will increment the version number on release:

- [x] Fix

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-2058

## :book: Description

Updated the postcss to wrap addition calculations in calc.

## :bulb: Context

A whole ton of issues with missing focus rings started popping up on beta today. There was one portion of the focus ring token with a calculation in it that wasn't wrapped in calc causing invalid css.

## :pencil: Checklist

<!--- Tick or place an `x` in all of the checkboxes that apply -->
<!--- Remove checkboxes that do not apply -->

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

## :crystal_ball: Next Steps

CP to beta

nikitadialpad commented:
hey Brad, it's some code alignment thing [here](https://deploy-preview-502--dialtone.netlify.app/design/illustrations/spot/). Could indentation be aligned like [it's on input page](https://deploy-preview-502--dialtone.netlify.app/components/input/)?

dialpad-drew commented on docs/design/illustrations/spot.html:
I read this as the assets _aren't_ theme-able by default, but can be _made_ theme-able by adding these classes into the asset. Is that intended?

dialpad-drew commented on docs/design/illustrations/spot.html:
Might recommend using this shortcode for inline code like the classes on these lines:  `{% code %}foo{% endcode %}`

dialpad-drew commented on docs/design/illustrations/spot.html:
`codeWellHeader` is typically for visualizing an example related to the code snippet. Is there a reason we're putting this copy in there and not with the other paragraphs? 

Also, see above about the `{% code%}` shortcode -- would recommend using it with the classes here.

dialpad-drew commented on docs/design/illustrations/spot.html:
What is this code snippet an example of? Theming? How a spot illustration would be implemented into UC/UV?

Also, nit: for code snippets, I'd recommend shift-tabbing so the left-most element is flush with the text editor window. These snippets area wrapped with `<pre>` and can preserve some funky spacing.

dialpad-drew reviewed (commented):
How does one _add_ a spot illustration to our project? I assume the SVG should be added to the project and the JSON file. Are there any additional or new gulp tasks that need to be run?

braddialpad commented on docs/design/illustrations/spot.html:
That's correct. Is it a possibility that some spot illustrations would look the same regardless of theme (say if the illustration was black and white) or do we want to enforce that all spot illustrations are themeable?

braddialpad commented on docs/design/illustrations/spot.html:
I just wanted it to be associated with the example, kind of like a caption, but I should probably change that since it's non standard across the rest of the site

braddialpad commented on docs/design/illustrations/spot.html:
yeah the indenting is messed, thanks for the eye.

this is an example of how to make your svg themeable when adding it to dialtone. I'm thinking maybe this should belong to the "adding icons" section of the site rather than here.

braddialpad commented:
> How does one _add_ a spot illustration to our project? I assume the SVG should be added to the project and the JSON file. Are there any additional or new gulp tasks that need to be run?

I see there is an adding icons page that splits icons into system/brand/patterns to some extent already. I'll add spot illustrations to here as well.

The process for adding a spot illustration is identical other than setting the `d-svg-primary--fill` and `d-svg-primary--stroke` classes if you want to use theme primary colors

dialpad-drew commented on docs/design/illustrations/spot.html:
If the spot illustration style uses our primary color (or derivates of our primary color), I'd suggest we make it theme-able by default. If not, I fear it undermines our claim that the system is theme-able and I'm not sure I see a benefit in making people manually manage SVG XML. Seems like it causes us to write more documentation and our consuming teams to read more documentation.

One could make the argument that Dialtone could provide an SVG component and we just start handling theme-ing in there. 😇

braddialpad commented on docs/design/illustrations/spot.html:
simple enough to do this if that's what we want, we can just replace those colors with the class on build.

braddialpad commented:
I've updated the gulp process to use the dialpad primary color for theming rather than a css class. 

Have also updated the adding icons page to include spot illustrations and added a section on theming. 

The spot illustrations page now links to the adding icons page to explain how to theme spot illustrations.

I believe all outstanding comments should be resolved.

dialpad-drew commented on docs/design/illustrations/spot.html:
Is it possible for just the spot illustration page's `.dialtone-icon-card__footer` to have `width: calc(var(--icon-card-width)*1.5)` set on it? Its current width is based on the asset being smaller icons, so with these assets being larger, it makes this pop-up area abnormally large. I'll show you a before and after:

## Before
<img width="633" alt="Screen Shot 2021-11-17 at 4 16 03 PM" src="https://user-images.githubusercontent.com/77751297/142297936-51e1e9d2-ec5d-41dc-b5ac-58597f31986d.png">

## After
<img width="548" alt="Screen Shot 2021-11-17 at 4 16 06 PM" src="https://user-images.githubusercontent.com/77751297/142297954-539efe00-1fe5-4030-b2e9-b9dfc61b1308.png">

dialpad-drew commented on docs/design/icons/adding.html:
I might recommend adding this _under_ the instructions. My theory is the instructions is the primary content people will seek more often and the theming bit is more secondary.

dialpad-drew reviewed (approved):
Two small things. Nice job, @braddialpad !

braddialpad commented:
> I might recommend adding this _under_ the instructions. My theory is the instructions is the primary content people will seek more often and the theming bit is more secondary. 

yep I agree, moved this to the bottom

braddialpad commented:
> Is it possible for just the spot illustration page's `.dialtone-icon-card__footer` to have `width: calc(var(--icon-card-width)*1.5)` set on it? Its current width is based on the asset being smaller icons, so with these assets being larger, it makes this pop-up area abnormally large. I'll show you a before and after:

done!

braddialpad committed:
fix(tokens): DLT-2058 fix uncalced focus ring token

juliodialpad commented on packages/dialtone-tokens/postcss/dialtone-tokens.cjs:
Why this was -1 previously? 🤔

juliodialpad reviewed (approved):
Fix looks good, just one question, thanks!

braddialpad commented on packages/dialtone-tokens/postcss/dialtone-tokens.cjs:
I had to reverse the array to get it to work correctly in the shadow parsing function after recent changes, and I thought if I put it at the end of the array it would output at the bottom, but actual array order doesn't seem to affect the order postcss uses to output (probably uses a linked list or something internally). This will now output after the shadow declarations instead of the top of the list, so it makes more logical sense when you are looking at the file.

No functional difference since order of css vars doesn't matter logic wise.
